hypervisor/dm/vpci/vpci.c

```diff
@@ -340,6 +340,13 @@ static int32_t vpci_write_pt_dev_cfg(struct pci_vdev *vdev, uint32_t offset,
 	} else {
 		/* passthru to physical device */
 		pci_pdev_write_cfg(vdev->pdev->bdf, offset, bytes, val);
+
+		/* Write back BAR HPA when doing FLR */
+		if (vdev->has_flr && (vdev->pcie_capoff + PCIR_PCIE_DEVCTRL == offset) &&
```

Comment:
just note if there is no alignment access which covered FLR here.

Reply:
Yes.

Reply:
Eventually we should decouple physical / virtual FLR... I suggest we decouple now, or add explicit TODO..